Source/WebCore/style/ClassChangeInvalidation.cpp

```diff
@@ -115,15 +115,27 @@ void ClassChangeInvalidation::computeInvalidation(const SpaceSplitString& oldCla
         m_element.invalidateStyle();
 
     auto invalidateBeforeChange = [](ClassChangeType type, IsNegation isNegation, MatchElement matchElement) {
-        if (matchElement == MatchElement::AnySibling || matchElement == MatchElement::HasNonSubjectOrScopeBreaking)
+        switch (matchElement) {
+        case MatchElement::AnySibling:
+        case MatchElement::ParentAnySibling:
+        case MatchElement::AncestorAnySibling:
+        case MatchElement::HasNonSubjectOrScopeBreaking:
             return true;
-        return type == ClassChangeType::Remove ? isNegation == IsNegation::No : isNegation == IsNegation::Yes;
+        default:
+            return type == ClassChangeType::Remove ? isNegation == IsNegation::No : isNegation == IsNegation::Yes;
+        }
     };
 
     auto invalidateAfterChange = [](ClassChangeType type, IsNegation isNegation, MatchElement matchElement) {
-        if (matchElement == MatchElement::AnySibling || matchElement == MatchElement::HasNonSubjectOrScopeBreaking)
+        switch (matchElement) {
+        case MatchElement::AnySibling:
+        case MatchElement::ParentAnySibling:
+        case MatchElement::AncestorAnySibling:
+        case MatchElement::HasNonSubjectOrScopeBreaking:
             return true;
-        return type == ClassChangeType::Add ? isNegation == IsNegation::No : isNegation == IsNegation::Yes;
+        default:
+            return type == ClassChangeType::Add ? isNegation == IsNegation::No : isNegation == IsNegation::Yes;
```

Comment:
It might be better to remove the default and list all values explicitly here so next person who adds a value has to consider this.